src/Sarif/Map/JsonMapBuilder.cs

```diff
@@ -15,58 +15,46 @@ namespace Microsoft.CodeAnalysis.Sarif.Map
-        public int MinimumSizeForNode { get; private set; }
-
-        /// <summary>
-        ///  Construct a JsonMapBuilder for the given target output size.
-        ///  (ex: 0.01 means 1% of the input file size)
-        /// </summary>
-        /// <param name="maxFileSizePercentage">Maximum size of the map relative to the source file (0.10 means 10%)</param>
-        public JsonMapBuilder(double maxFileSizePercentage)
-        {
-            if (maxFileSizePercentage <= 0 || maxFileSizePercentage >= 100) { throw new ArgumentOutOfRangeException("maxFileSizePercentage must be between 0 and 100."); }
-            MaxFileSizePercentage = maxFileSizePercentage;
-            MinimumSizeForNode = (int)(NodeSizeEstimateBytes / maxFileSizePercentage);
-        }
-
         /// <summary>
         ///  Build the JsonMap for a given Json file, by path.
         ///  Returns null if the source file was too small for any map nodes to fit the size budget.
         /// </summary>
         /// <param name="filePath">File Path to the Json file to build a map from</param>
+        /// <param name="settings">JsonMapSettings for map; null for defaults</param>
         /// <returns>JsonMap for file or null if file too small for map</returns>
-        public JsonMapNode Build(string filePath)
+        public static JsonMapNode Build(string filePath, JsonMapSettings settings = null)
         {
-            return Build(() => File.OpenRead(filePath));
+            return Build(() => File.OpenRead(filePath), settings);
         }
 
         /// <summary>
         ///  Build the JsonMap for a given Json file, given a stream provider.
         ///  Returns null if the source file was too small for any map nodes to fit the size budget.
         /// </summary>
         /// <param name="streamProvider">A function which will open the stream for the desired file</param>
+        /// <param name="settings">JsonMapSettings for map; null for defaults</param>
         /// <returns>JsonMap for file or null if file too small for map</returns>
-        public JsonMapNode Build(Func<Stream> streamProvider)
+        public static JsonMapNode Build(Func<Stream> streamProvider, JsonMapSettings settings = null)
         {
+            if (settings == null) { settings = JsonMapSettings.DefaultSettings; }
+
             using (Stream stream = streamProvider())
             {
-                if (stream.Length <= MinimumSizeForNode) { return null; }
+                long length = stream.Length;
+                if (length <= settings.MinimumSizeForNode) { return null; }
+                settings.AdjustForFileSize(length);
             }
 
             using (JsonPositionedTextReader reader = new JsonPositionedTextReader(streamProvider))
             {
                 if (!reader.Read()) { return null; }
-                return Build(reader, 0, out long unused);
+                return Build(reader, settings, 0, out long unused);
```

Comment:
>out long unused [](start = 50, length = 15)

C #7 has a lovely idiom for unused ("discarded") `out` parameters:
```
return Build(reader, settings, 0, out _);
```

Reply:
Huh. I actually prefer "unused" because it's self-documenting for people unfamiliar with that feature, but I've changed it as you suggest.